data_steward/cdr_cleaner/manual_cleaning_rules/remediate_basics.py

```diff
@@ -0,0 +1,395 @@
+"""
+This script updates OBSERVATION, SURVEY_CONDUCT, and PERSON tables with the
+given new set of the basics survey responses.
+
+This is a one-off cleaning rule for DC-3016 hot-fix. It is not expected to
+run for every CDR release.
+
+Original Issues: DC-3016 and its subtasks.
+"""
+
+#Python imports
+import logging
+
+# Project imports
+from common import (JINJA_ENV, OBSERVATION, PERSON, SURVEY_CONDUCT)
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+from constants.cdr_cleaner.clean_cdr import (COMBINED,
+                                             CONTROLLED_TIER_DEID_BASE,
+                                             CONTROLLED_TIER_DEID_CLEAN, QUERY,
+                                             REGISTERED_TIER_DEID_BASE,
+                                             REGISTERED_TIER_DEID_CLEAN)
+from resources import ext_table_for, mapping_table_for
+from retraction.retract_utils import is_combined_dataset, is_deid_dataset
+
+LOGGER = logging.getLogger(__name__)
+
+ISSUE_NUMBERS = ['DC3016']
+
+CREATE_NEW_OBS_ID_LOOKUP = JINJA_ENV.from_string("""
+CREATE TABLE IF NOT EXISTS `{{project}}.{{sandbox_dataset}}.{{new_obs_id_lookup}}`
+(
+    source_observation_id INT64 NOT NULL OPTIONS(description="observation_id from the source table."),
+    dataset_id STRING NOT NULL OPTIONS(description="Dataset ID of the destination table."),
+    observation_id INT64 NOT NULL OPTIONS(description="Newly assigned observation_id for the destination dataset.")
+)
+OPTIONS
+(
+    description="Lookup table for observation_ids. New observation_ids are assigned when loaded so no observation_id will be a duplicate."
+)
+""")
+
+# NOTE In case of re-running remediation for the same dataset, you need to run
+# the following delete statement beforehand:
+# DELETE * FROM new_obs_id_lookup WHERE dataset='the dataset you need to re-run on'
+# Otherwise, there will be duplicate entries in the lookup table and in OBS,
+# OBS_EXT, and OBS_MAPPING tables in the output.
+INSERT_NEW_OBS_ID_LOOKUP = JINJA_ENV.from_string("""
+INSERT INTO `{{project}}.{{sandbox_dataset}}.{{new_obs_id_lookup}}`
+(source_observation_id, dataset_id, observation_id)
+SELECT
+    inc_o.observation_id,
+    '{{dataset}}',
+    ROW_NUMBER() OVER(ORDER BY inc_o.observation_id
+        ) + (
+        SELECT MAX(observation_id) 
+        FROM `{{project}}.{{dataset}}.{{table}}`
+        )
+FROM `{{project}}.{{incremental_dataset}}.{{table}}` inc_o
+WHERE person_id IN (
+    SELECT person_id FROM `{{project}}.{{dataset}}.{{table}}` 
+)
+""")
+
+SANDBOX_OBS = JINJA_ENV.from_string("""
```

Comment:
Imagined scenario:
Person A answers parent question and 1 or more child questions in bad dataset.
Person A has data in incremental dataset and answers parent question differently, such that they should not have answered the child questions from their previous parent answer.
Because the sandboxing is done on `person_id` and `observation_source_concept_id`, the old child data will not be sandboxed.  

Is this a risk we are comfortable with or should we modify the sandboxing to use the `concept_id`list (to be generated to the best of our abilities knowing there is an issue with health insurance related questions)?

@mandyroberts

Reply:
@lrwb-aou & @molly-giddings will look at how many person_ids might have such a scenario in the data.

Reply:
@lrwb-aou put the SQL in this ticket that identifies data of this scenario.  The results show consistent multi-select question value pairs for all data.  Therefore, there does not seem to be a risk of creating non-sandboxed data for participants with existing real data.